core/lib/Cardano/Address/Errors.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE ExistentialQuantification #-}
+
+-- |
+-- Copyright: © 2020 IOHK
+-- License: Apache-2.0
+
+module Cardano.Address.Errors
+    ( -- $overview
+
+      -- * AddrError
+      AddrError(..)
+  ) where
+
+import Prelude
+
+import Cardano.Address
+    ( Address )
+
+import Control.Exception
+    ( Exception )
+
+-- $overview
+--
+-- This module provides error/exception types for Addresses.
+
+data AddrError = UnknownAddrType Address
+               | WrongInputSize Address
+                                Int -- the actual size
+               | PtrRetrieveError Address
+                                  String -- human readable error of underlying operation
+               | InvalidDerivationPath Address
+               | forall e . (Exception e, Show e) => DeserialiseError Address e
```

Comment:
Avoid variable-length indentation whenever possible. It plays poorly with git diff. Consider instead:

```hs
data AddrError
    = UnknownAddrType Address
    | WrongInputSize Address Int 
    | PtrRetrieveError Address String
    | InvalidDerivationPath Address 
    | forall e. (Exception e, Show e) => DeserialiseError Address e
```

Reply:
see also: https://github.com/input-output-hk/adrestia/wiki/Coding-Standards#avoid-variable-length-indentation